source/guides/azure/infiniband.md

```diff
@@ -0,0 +1,199 @@
+```
+ip addr show
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+    inet6 ::1/128 scope host 
+       valid_lft forever preferred_lft forever
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
+    link/ether 60:45:bd:a7:42:cc brd ff:ff:ff:ff:ff:ff
+    inet 10.6.0.5/24 brd 10.6.0.255 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::6245:bdff:fea7:42cc/64 scope link 
+       valid_lft forever preferred_lft forever
+3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
+    link/ether 00:15:5d:33:ff:16 brd ff:ff:ff:ff:ff:ff
+4: enP44906s1: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc mq master eth0 state UP group default qlen 1000
+    link/ether 60:45:bd:a7:42:cc brd ff:ff:ff:ff:ff:ff
+    altname enP44906p0s2
+5: ibP59423s2: <BROADCAST,MULTICAST> mtu 4092 qdisc noop state DOWN group default qlen 256
+    link/infiniband 00:00:09:27:fe:80:00:00:00:00:00:00:00:15:5d:ff:fd:33:ff:16 brd 00:ff:ff:ff:ff:12:40:1b:80:1d:00:00:00:00:00:00:ff:ff:ff:ff
+    altname ibP59423p0s2
+
+$ nvidia-smi topo -m
+	GPU0	GPU1	GPU2	GPU3	GPU4	GPU5	GPU6	GPU7	CPU Affinity	NUMA Affinity
+GPU0	 X 	NV2	NV1	NV2	NODE	NODE	NV1	NODE	0-19	0
+GPU1	NV2	 X 	NV2	NV1	NODE	NODE	NODE	NV1	0-19	0
+GPU2	NV1	NV2	 X 	NV1	NV2	NODE	NODE	NODE	0-19	0
+GPU3	NV2	NV1	NV1	 X 	NODE	NV2	NODE	NODE	0-19	0
+GPU4	NODE	NODE	NV2	NODE	 X 	NV1	NV1	NV2	0-19	0
+GPU5	NODE	NODE	NODE	NV2	NV1	 X 	NV2	NV1	0-19	0
+GPU6	NV1	NODE	NODE	NODE	NV1	NV2	 X 	NV2	0-19	0
+GPU7	NODE	NV1	NODE	NODE	NV2	NV1	NV2	 X 	0-19	0
+
+Legend:
+
+  X    = Self
+  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
+  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
+  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
+  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
+  PIX  = Connection traversing at most a single PCIe bridge
+  NV#  = Connection traversing a bonded set of # NVLinks
+
+```
+
+## Install UCX-Py and tools
+```
+$ wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+$ bash Miniconda3-latest-Linux-x86_64.sh
+```
+Accept the default and allow conda init to run. Then start a new shell.
+
+Create a conda environment (see ucx-py docs)
+
+```
+$ conda create -n ucxpy -c conda-forge -c rapidsai python=3.7 ipython ucx-proc=*=gpu ucx ucx-py dask distributed numpy cupy pytest pynvml -y
+...
+$ conda activate ucxpy
+```
+Clone ucx-py repo locally
+
+```
+git clone https://github.com/rapidsai/ucx-py.git
+cd ucx-py/benchmarks
+```
+
+## Run benchmarks
+[GitHub Issue on Benchmarks](https://github.com/rapidsai/ucx-py/issues/311)
```

Comment:
Looks like there have been changes to the benchmarks and tests since the @jacobtomlinson's tests and these instructions.

@quasiben @pentschev - Do you know the best way to test UCXPY? Just a single instance with IB setup, for now.

Reply:
I would primarily suggest running the UCX-Py test suite, from within the `ucx-py` repo:

```bash
pytest -vs tests/
pytest -vs ucp/_libs/tests/
```

The second piece would be checking whether IB works, for that you can run some of the benchmarks that we include, for example:

```bash
# Let UCX pick the best transport (expecting NVLink when available,
# otherwise IB, or TCP in worst case) on devices 0 and 1
python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB

# Force TCP-only on devices 0 and 1
UCX_TLS=tcp,cuda_copy python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB
```

We expect the first case above to have much higher bandwidth than the second. If you happen to have both NVLink and IB connectivity, then you may limit to the specific transport by specifying `UCX_TLS`, e.g.:

```bash
# NVLink (if available) or TCP
UCX_TLS=tcp,cuda_copy,cuda_ipc

# IB (if available) or TCP
UCX_TLS=tcp,cuda_copy,rc
```

Finally, I would suggest running Dask-CUDA cuDF benchmark:

```bash
# Default Dask TCP communication protocol
python -m dask_cuda.benchmarks.local_cudf_merge --devs 0,1,2,3,4,5,6,7
 --chunk-size 100_000_000

# UCX protocol
python -m dask_cuda.benchmarks.local_cudf_merge --devs 0,1,2,3,4,5,6,7
 --chunk-size 100_000_000 --protocol ucx
```

Above we're running for devices 0..7, adjust that for the number of devices available, the default is to run on GPU 0 only. Also `--chunk-size 100_000_000` is a safe value for 32GB GPUs, you may adjust that proportional to the size of the GPU you have (it scales linearly, so `50_000_000` should be good for 16GB or `150_000_000` for 48GB). Please note that TCP will be _much_ slower, so you may choose to adjust to the maximum size you can get for UCX to see how far you can push it, but you may want to run on a much smaller workload such as `10_000_000` to compare TCP and UCX.

Reply:
Thanks, @pentschev This is working as expected. I appreciate the help!

@jacobtomlinson I'll tidy up these instructions and we can move forward finalizing this PR.